packages/access-client/src/access.js

```diff
@@ -0,0 +1,335 @@
+    audience: audience = agent.did(),
+    access = spaceAccess,
+  }
+) => {
+  // Request access from the account.
+  const { out: result } = await agent.invokeAndExecute(Access.authorize, {
+    audience: DID.parse(provider),
+    with: audience,
+    nb: {
+      iss: account,
+      // New ucan spec moved to recap style layout for capabilities and new
+      // `access/request` will use similar format as opposed to legacy one,
+      // in the meantime we translate new format to legacy format here.
+      att: [...toCapabilities(access)],
+    },
+  })
+
+  return result.error
+    ? result
+    : {
+        ok: new PendingAccessRequest({
+          ...result.ok,
+          agent,
+          audience,
+          provider,
+        }),
+      }
+}
+
+/**
+ * Claims access that has been delegated to the given audience, which by
+ * default is the agent's DID.
+ *
+ * @param {API.Agent} agent
+ * @param {object} input
+ * @param {API.DID} [input.audience] - Principal requesting an access.
+ * @param {API.ProviderDID} [input.provider] - Provider handling the invocation.
+ * @returns {Promise<API.Result<GrantedAccess, API.AccessClaimFailure|API.InvocationError>>}
+ */
+export const claim = async (
+  agent,
+  {
+    provider = /** @type {API.ProviderDID} */ (agent.connection.id.did()),
+    audience = agent.did(),
+  } = {}
+) => {
+  const { out: result } = await agent.invokeAndExecute(Access.claim, {
+    audience: DID.parse(provider),
+    with: audience,
+  })
+
+  if (result.error) {
+    return result
+  } else {
+    const delegations = Object.values(result.ok.delegations)
+    const proofs = delegations.flatMap((proof) => bytesToDelegations(proof))
+    return { ok: new GrantedAccess({ agent, provider, audience, proofs }) }
+  }
+}
+
+/**
+ * Represents a pending access request. It can be used to poll for the requested
+ * delegation.
+ */
+class PendingAccessRequest {
+  /**
+   * @typedef {object} PendingAccessRequestModel
+   * @property {API.Agent} source.agent - Agent handling interaction.
+   * @property {API.DID} source.audience - Principal requesting an access.
+   * @property {API.ProviderDID} source.provider - Provider handling request.
+   * @property {API.UTCUnixTimestamp} source.expiration - Seconds in UTC.
+   * @property {API.Link} source.request - Link to the `access/authorize` invocation.
+   *
+   * @param {PendingAccessRequestModel} model
+   */
+  constructor(model) {
+    this.model = model
+  }
+
+  get agent() {
+    return this.model.agent
+  }
+  get audience() {
+    return this.model.audience
+  }
+  get expiration() {
+    return new Date(this.model.expiration * 1000)
+  }
+
+  get request() {
+    return this.model.request
+  }
+
+  get provider() {
+    return this.model.provider
+  }
+
+  /**
+   *
+   * @returns {Promise<API.Result<API.Delegation[], API.InvocationError|API.AccessClaimFailure|RequestExpired>>}
+   */
+  async poll() {
```

Comment:
How should this class be used? Should I call `poll` or `claim` on instances I receive? The docs above say I can use this to poll for the requested delegation but I think I would need to call `claim` not `poll` for that behaviour?

It seems back to front to me that `poll` sends a single request, but `claim` sends multiple requests. Perhaps `poll` should be a private method?